src/charm.py

```diff
@@ -530,11 +578,67 @@ def _add_log_forwarding(self, endpoints: List[dict], insecure_skip_verify: bool)
                 pipelines=["logs"],
             )
 
+    def _add_traces_ingestion(self, requested_tracing_protocols: Set[ReceiverProtocol]):
+        """Configure the tracing receivers for otel-collector to ingest traces.
+
+        Args:
+            requested_tracing_protocols: The tracing protocols for which to enable receivers.
+        """
+        # TODO: check with the team, do we keep this?
+        # TODO: should we just add the otlp protocols always? probably yes
+        if not requested_tracing_protocols:
+            logger.warning("No tempo receivers enabled: otel-collector cannot ingest traces.")
+            return
+
+        if "zipkin" in requested_tracing_protocols:
+            self.otel_config.add_receiver(
+                name="zipkin",
+                receiver_config={"endpoint": f"0.0.0.0:{PORTS.ZIPKIN}"},
+                pipelines=["traces"],
+            )
+        if (
+            "jaeger_grpc" in requested_tracing_protocols
+            or "jaeger_thrift_http" in requested_tracing_protocols
+        ):
+            jaeger_config = {"protocols": {}}
+            if "jaeger_grpc" in requested_tracing_protocols:
+                jaeger_config["protocols"].update(
+                    {"grpc": {"endpoint": f"0.0.0.0:{PORTS.JAEGER_GRPC}"}}
+                )
+            if "jaeger_thrift_http" in requested_tracing_protocols:
+                jaeger_config["protocols"].update(
+                    {"thrift_http": {"endpoint": f"0.0.0.0:{PORTS.JAEGER_THRIFT_HTTP}"}}
+                )
+            self.otel_config.add_receiver(
+                name="jaeger", receiver_config=jaeger_config, pipelines=["traces"]
+            )
```

Comment:
```suggestion
            if jaeger_config:
                self.otel_config.add_receiver(
                    name="jaeger", receiver_config=jaeger_config, pipelines=["traces"]
                )
```

But as mentioned earlier, I wonder if we'd rather not support anything other than otlp.